src/Translator.php

```diff
@@ -126,11 +126,44 @@ private function selectRightPluralForm(string $translation, string $lang, int $c
         if (count($exploded) === 1) {
             return $translation;
         }
-
+        foreach ($exploded as $value) {
+            $translationPlural = $this->findSpecialFormat($value, $count);
+            if ($translationPlural) {
+                return $translationPlural;
+            }
+        }
         $pluralForm = PluralForm::get($count, $lang);
         return isset($exploded[$pluralForm]) ? $exploded[$pluralForm] : $exploded[0];
     }
 
+    /**
+     * example: '{0} používateľov|{1}používateľ|[2,4] používatelia|[5,Inf] používateľov';
+     * @param string $translationForm
+     * @param int $count
+     * @return string|null
+     */
+    private function findSpecialFormat(string $translationForm, int $count): ?string
```

Comment:
This implementation is very narrowed. What about open/close intervals like Symfony translator does:

`{0}%name% has no apples|{1}%name% has one apple|]1,Inf[ %name% has %count% apples`

More here https://symfony.com/doc/3.1/components/translation/usage.html#pluralization

Reply:
Seems better but too much regexp and explodes to me. Don't you want to add some unit test to prove it right?

Reply:
nice! 👍